### ISSUE TITLE
linter: typescript/no-invalid-void-type does not recognize generic parameter as valid

### ISSUE THREAD

**Issue by rajzik** [oxc-project/oxc#19541]
### What version of Oxlint are you using?

1.48.0

### What command did you run?

oxlint --type-aware --type-check --deny-warnings --report-unused-disable-directives

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "plugins": ["typescript"],
  "rules": {
     "typescript/no-invalid-void-type": "error"
  }
}
```

### What happened?

When linting following code: 
```ts
// Use `void` only as a return type or generic type argument.
// help: Replace this `void` type with an allowed type, or keep `void` only in a valid return position.
builder.query<AppConfig, void>()
```
Void type is not recognized as valid generic value which from the documentation is.

**Comment by camc314** [comment]
I get the same error when using the typescript eslint versio of this? https://typescript-eslint.io/play/#ts=5.9.2&fileType=.ts&code=PTAEFUGcFNQAwG4HsCWATOokDsA2BPUAQ0mNACdoAXAV3O1CvwAdYlzQBzabaclAMaMWsIuU40AtjyoA6AFAhQAC2i5mALlAAlaM1xEBsKspSlEqDMNagA7ihPEGRXLiS3oaa9AA0WDgDW0HrwyOiYOASgKM6gCC7oFNR0DMxIkA4oOAoARjQouGh8sgCONHz4ADwAgszMAMI4AGYonH5haAB8ABQAlEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uYDcBDRgBNaPDpWFFS6KImjQO0SODABfECqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

**Comment by camc314** [comment]
Closing as not planned - we appear to have the same behaviour as typescript-eslint.

Please comment/reopen if this is not the case - thanks!

**Comment by rajzik** [comment]
I think they have open issue for this error: https://github.com/typescript-eslint/typescript-eslint/issues/8113 
Maybe this can be reopened and fixed in oxc as well? But I am not sure, if OXC plans to extend or enhance linter rules or just copy existing ones?

**Comment by rajzik** [comment]
@camc314 Sorry about ping, but I think there is no notification when issue is closed, can you chech my previous message?
